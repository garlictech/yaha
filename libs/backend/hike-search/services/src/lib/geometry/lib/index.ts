export * from './circle-slice';
export * from './coordinate-iterator';
export * from './line-calculations';
export * from './point-transformations';
export * from './box-calculations';
export * from './polygon-calculations';
export * from './geospatial-calculations';
export * from './filter-points-close-to-reference-points';
export * from './hike-calculations';
export * from './circle-fp';
