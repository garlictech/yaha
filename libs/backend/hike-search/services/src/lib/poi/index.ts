export * from './lib/poi.fp';
