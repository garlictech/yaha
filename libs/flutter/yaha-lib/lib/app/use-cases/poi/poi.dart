export 'poi.dart';
export 'end_points_of_hike.dart';
export 'touristic_pois_along_hike_with_dynamic_pois.dart';
export 'poi_provider.dart';
export 'pois_along_hike.dart';
export 'random_touristic_pois_along_hike.dart';
export 'touristic_pois_along_hike.dart';
export 'touristic_pois_along_hike_sorted_by_distance.dart';
export 'filtered_pois_along_hike.dart';
export 'filtered_pois_around_hike.dart';
