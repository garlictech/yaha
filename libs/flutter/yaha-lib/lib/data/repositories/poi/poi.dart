export 'poi-repository-neo4j.dart';
