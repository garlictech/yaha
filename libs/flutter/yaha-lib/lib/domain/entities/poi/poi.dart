export 'poi_type.dart';
export 'poi_entity.dart';
export 'supported_pois.dart';
export 'poi_of_hike.dart';
