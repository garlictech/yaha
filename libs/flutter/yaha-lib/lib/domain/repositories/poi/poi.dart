export 'poi_repository.dart';
