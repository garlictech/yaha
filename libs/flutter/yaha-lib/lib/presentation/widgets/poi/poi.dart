export 'poi-title-list.dart';
export 'poi-icon.dart';
export 'poi-filters-list.dart';
export 'things_on_route_as_icons.dart';
export 'interesting_places_on_route_list.dart';
