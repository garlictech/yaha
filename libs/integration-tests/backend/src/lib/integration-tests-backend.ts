export function integrationTestsBackend(): string {
  return 'integration-tests-backend';
}
