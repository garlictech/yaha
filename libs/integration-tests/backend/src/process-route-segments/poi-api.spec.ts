import { NestFactory } from '@nestjs/core';
import { tap, map, switchMap } from 'rxjs/operators';
import { PoiApiService, PoiModule } from '@bit/garlictech.nestjs.gtrack.poi';

import { poiFixture, getIdOfPoi } from './fixtures';
import * as E from 'fp-ts/lib/Either';
import { Poi } from '@bit/garlictech.universal.gtrack.graphql-api';

describe('POI graphql integration tests', () => {
  let poiApiService: PoiApiService;

  beforeAll(async () => {
    const app = await NestFactory.createApplicationContext(PoiModule);
    poiApiService = app.get(PoiApiService);
  });

  test('it must be able to CRUD pois', done => {
    poiApiService.api
      .delete(getIdOfPoi(poiFixture()))
      .pipe(
        switchMap(() => poiApiService.api.add(poiFixture())),
        tap(result => expect(result).toMatchSnapshot('ADD')),
        map(E.fold(done.fail, res => res.id)),
        switchMap((id: string) => poiApiService.api.getById(id)),
        tap(result => expect(result).toMatchSnapshot('GETBYID')),
        map(E.fold(done.fail, res => res.id)),
        switchMap((id: string) => poiApiService.api.getMultipleItemsById([id])),
        tap(result => expect(result).toMatchSnapshot('GETMULTIPLEITEMSBYID')),
        map(E.fold(done.fail, res => res[0].id)),
        switchMap((id: string) =>
          poiApiService.api.update(id, { address: 'foobar addres 2' }),
        ),
        tap(result => expect(result).toMatchSnapshot('UPDATE')),
        map(E.fold(done.fail, (res: Poi) => res.id)),
        switchMap((id: string) =>
          poiApiService.api.updateMultipleItems([
            { id, address: 'foobar addres 3' },
          ]),
        ),
        tap(result => expect(result).toMatchSnapshot('UPDATEMULTIPLEITEMS')),
        map(E.fold(done.fail, (res: Poi[]) => res[0].id)),
        switchMap((id: string) => poiApiService.api.delete(id)),
        tap(result => expect(result).toMatchSnapshot('DELETE')),
      )
      .subscribe(
        () => done(),
        err => done.fail(err),
      );
  }, 20000);
});
