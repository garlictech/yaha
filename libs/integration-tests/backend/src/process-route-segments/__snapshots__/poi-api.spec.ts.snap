// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POI graphql integration tests it must be able to CRUD pois: ADD 1`] = `
Object {
  "_tag": "Right",
  "right": Object {
    "__typename": "Poi",
    "address": null,
    "description": Array [
      Object {
        "__typename": "TextualDescription",
        "fullDescription": null,
        "languageKey": "en_US",
        "summary": null,
        "title": "Waste basket",
        "type": "markdown",
      },
    ],
    "elevation": 519,
    "id": "47.505118-18.94385",
    "lat": 47.505118,
    "lon": 18.94385,
    "openingHours": null,
    "phoneNumber": null,
    "sourceObject": Array [
      Object {
        "__typename": "PoiSourceObject",
        "languageKey": "en_US",
        "objectId": "4340332291",
        "objectType": "osmAmenity",
        "url": null,
      },
    ],
    "tags": null,
    "types": Array [
      "waste_basket",
    ],
  },
}
`;

exports[`POI graphql integration tests it must be able to CRUD pois: DELETE 1`] = `
Object {
  "_tag": "Right",
  "right": true,
}
`;

exports[`POI graphql integration tests it must be able to CRUD pois: GETBYID 1`] = `
Object {
  "_tag": "Right",
  "right": Object {
    "__typename": "Poi",
    "address": null,
    "description": Array [
      Object {
        "__typename": "TextualDescription",
        "fullDescription": null,
        "languageKey": "en_US",
        "summary": null,
        "title": "Waste basket",
        "type": "markdown",
      },
    ],
    "elevation": 519,
    "id": "47.505118-18.94385",
    "lat": 47.505118,
    "lon": 18.94385,
    "openingHours": null,
    "phoneNumber": null,
    "sourceObject": Array [
      Object {
        "__typename": "PoiSourceObject",
        "languageKey": "en_US",
        "objectId": "4340332291",
        "objectType": "osmAmenity",
        "url": null,
      },
    ],
    "tags": null,
    "types": Array [
      "waste_basket",
    ],
  },
}
`;

exports[`POI graphql integration tests it must be able to CRUD pois: GETMULTIPLEITEMSBYID 1`] = `
Object {
  "_tag": "Right",
  "right": Array [
    Object {
      "__typename": "Poi",
      "address": null,
      "description": Array [
        Object {
          "__typename": "TextualDescription",
          "fullDescription": null,
          "languageKey": "en_US",
          "summary": null,
          "title": "Waste basket",
          "type": "markdown",
        },
      ],
      "elevation": 519,
      "id": "47.505118-18.94385",
      "lat": 47.505118,
      "lon": 18.94385,
      "openingHours": null,
      "phoneNumber": null,
      "sourceObject": Array [
        Object {
          "__typename": "PoiSourceObject",
          "languageKey": "en_US",
          "objectId": "4340332291",
          "objectType": "osmAmenity",
          "url": null,
        },
      ],
      "tags": null,
      "types": Array [
        "waste_basket",
      ],
    },
  ],
}
`;

exports[`POI graphql integration tests it must be able to CRUD pois: UPDATE 1`] = `
Object {
  "_tag": "Right",
  "right": Object {
    "__typename": "Poi",
    "address": "foobar addres 2",
    "description": Array [
      Object {
        "__typename": "TextualDescription",
        "fullDescription": null,
        "languageKey": "en_US",
        "summary": null,
        "title": "Waste basket",
        "type": "markdown",
      },
    ],
    "elevation": 519,
    "id": "47.505118-18.94385",
    "lat": 47.505118,
    "lon": 18.94385,
    "openingHours": null,
    "phoneNumber": null,
    "sourceObject": Array [
      Object {
        "__typename": "PoiSourceObject",
        "languageKey": "en_US",
        "objectId": "4340332291",
        "objectType": "osmAmenity",
        "url": null,
      },
    ],
    "tags": null,
    "types": Array [
      "waste_basket",
    ],
  },
}
`;

exports[`POI graphql integration tests it must be able to CRUD pois: UPDATEMULTIPLEITEMS 1`] = `
Object {
  "_tag": "Right",
  "right": Array [
    Object {
      "__typename": "Poi",
      "address": "foobar addres 3",
      "description": Array [
        Object {
          "__typename": "TextualDescription",
          "fullDescription": null,
          "languageKey": "en_US",
          "summary": null,
          "title": "Waste basket",
          "type": "markdown",
        },
      ],
      "elevation": 519,
      "id": "47.505118-18.94385",
      "lat": 47.505118,
      "lon": 18.94385,
      "openingHours": null,
      "phoneNumber": null,
      "sourceObject": Array [
        Object {
          "__typename": "PoiSourceObject",
          "languageKey": "en_US",
          "objectId": "4340332291",
          "objectType": "osmAmenity",
          "url": null,
        },
      ],
      "tags": null,
      "types": Array [
        "waste_basket",
      ],
    },
  ],
}
`;
